programs/mango-v4/src/health/client.rs

```diff
@@ -446,6 +446,58 @@ impl HealthCache {
             cache.health_ratio(HealthType::Init)
         })
     }
+
+    /// Computes the account equity and the summarized balance sheet marked to market.
+    /// Returns (equity, assets, liabilities)
+    pub fn equity(&self) -> (I80F48, I80F48, I80F48) {
+        let mut assets = I80F48::ZERO;
+        let mut liabs = I80F48::ZERO;
+
+        for token_info in self.token_infos.iter() {
+            if token_info.balance_spot.is_negative() {
+                liabs -= token_info.balance_spot * token_info.prices.oracle;
+            } else {
+                assets += token_info.balance_spot * token_info.prices.oracle;
+            }
+        }
+
+        for serum_info in self.serum3_infos.iter() {
+            let quote = &self.token_infos[serum_info.quote_info_index];
+            let base = &self.token_infos[serum_info.base_info_index];
+            assets += serum_info.reserved_base * base.prices.oracle;
+            assets += serum_info.reserved_quote * quote.prices.oracle;
+        }
+
+        for perp_info in self.perp_infos.iter() {
+            // TODO: multiply with quote price
```

Comment:
yeah, multiply both the quote and base contributions with the price for perp_info.settle_token_index

Reply:
why base? base oracle should be base/USD no?

Reply:
It's the way perp margining and perp settlement works right now. The oracle is base/USD, but the settlement is in USDC. If you have 1 base long and the price goes up by 1 USD then you can settle 1 USDC. Hence the USDC price needs to be applied to the base oracle here.

We talked about switching from "perps on SOL/USD with USDC payouts" to "perps on SOL/USDC perps with USDC payouts" because the latter should be more interesting to users when the USDC price is unstable. But that has not yet been done.